src/java/org/apache/cassandra/db/ReadExecutionController.java

```diff
@@ -134,10 +134,14 @@ static ReadExecutionController forCommand(ReadCommand command)
         }
     }
 
-    private static ColumnFamilyStore maybeGetIndexCfs(ColumnFamilyStore baseCfs, ReadCommand command)
+    private static ColumnFamilyStore maybeGetIndexCfs(ReadCommand command)
     {
-        Index index = command.getIndex(baseCfs);
-        return index == null ? null : index.getBackingTable().orElse(null);
+        Index.QueryPlan queryPlan = command.indexQueryPlan();
+        if (queryPlan == null)
+            return null;
+
+        // only the index groups with a single member are allowed to have a backing table
+        return queryPlan.getFirst().getBackingTable().orElse(null);
```

Comment:
Just an aside, this feels like a place where `Optional` isn't particularly useful. All we do is translate from CFS/null in, for instance `CassandraIndex`, to an `Optional`, back to `CFS/null` again in `forCommand()` above (where we proceed to check for null anyway). Again, don't feel the need to change this :)